File: sdk-typescript/examples/webhook-management-example.ts
// To run this example:
// 1. Ensure you have the SDK installed: npm install pearl-sdk
// 2. Transpile and run:
//    npx ts-node examples/webhook-management-example.ts

import { PearlClient, WebhookEndpointRequest } from 'pearl-sdk';
import axios from 'axios'; // Import axios for error checking

// --- IMPORTANT: Configure your API Key and Webhook URL ---
// Replace 'YOUR_PEARL_API_KEY' with your actual Pearl API Key.
// Replace 'YOUR_WEBHOOK_URL' with the actual URL of your webhook endpoint.
const API_KEY: string = 'YOUR_PEARL_API_KEY'; // Explicitly type as string
const YOUR_WEBHOOK_URL: string = 'https://your-app.com/api/pearl-webhooks'; // Explicitly type as string

async function runWebhookManagementExample() {
  if (API_KEY === 'YOUR_PEARL_API_KEY') {
    console.warn("WARNING: Please replace 'YOUR_PEARL_API_KEY' with your actual API key.");
    console.log("Exiting example as API Key is a placeholder.");
    return;
  }
  if (YOUR_WEBHOOK_URL === 'https://your-app.com/api/pearl-webhooks') {
    console.warn("WARNING: Please replace 'YOUR_WEBHOOK_URL' with your actual webhook endpoint URL.");
    console.log("Exiting example as Webhook URL is a placeholder.");
    return;
  }

  const client = new PearlClient({ apiKey: API_KEY });

  try {
    console.log("\n--- Starting Webhook Management Example ---");

    // --- Register Webhook Endpoint ---
    console.log(`Attempting to register webhook endpoint: ${YOUR_WEBHOOK_URL}`);
    const registerRequest: WebhookEndpointRequest = { endpoint: YOUR_WEBHOOK_URL };
    await client.webhooks.register(registerRequest);
    console.log(`Successfully registered webhook endpoint.`);

    // --- Update Webhook Endpoint (Example) ---
    // If your webhook URL changes, you can use the update method.
    // For this example, we'll simulate an update to a slightly different path.
    // By explicitly typing YOUR_WEBHOOK_URL as string, TypeScript won't narrow it to 'never'.
    const updatedWebhookUrl = `${YOUR_WEBHOOK_URL.replace('/pearl-webhooks', '/v2/pearl-webhooks')}`;
    console.log(`\nAttempting to update webhook endpoint to: ${updatedWebhookUrl}`);
    const updateRequest: WebhookEndpointRequest = { endpoint: updatedWebhookUrl };
    await client.webhooks.update(updateRequest);
    console.log(`Successfully updated webhook endpoint.`);

  } catch (error: any) {
    console.error("\n--- Webhook Management API Error ---");
    if (axios.isAxiosError(error)) {
      console.error("Error message:", error.message);
      if (error.response) {
        console.error("Status:", error.response.status);
        console.error("Data:", JSON.stringify(error.response.data, null, 2));
      }
    } else {
      console.error("An unexpected error occurred:", error);
    }
  } finally {
    console.log("\n--- End of Webhook Management Example ---");
  }
}

// Run the example
runWebhookManagementExample();

File: sdk-python/examples/webhook_management_example.py
#!/usr/bin/env python3
"""
Webhook Management Example

This example demonstrates how to use the Pearl SDK to register and update webhook endpoints.

To run this example:
1. Install the SDK: pip install pearl-sdk
2. Set your API key and webhook URL in the variables below
3. Run: python examples/webhook_management_example.py
"""

from pearl_sdk import PearlClient, WebhookEndpointRequest

# --- IMPORTANT: Configure your API Key and Webhook URL ---
# Replace 'YOUR_PEARL_API_KEY' with your actual Pearl API Key.
# Replace 'YOUR_WEBHOOK_URL' with the actual URL of your webhook endpoint.
API_KEY = 'YOUR_PEARL_API_KEY'
YOUR_WEBHOOK_URL = 'https://your-app.com/api/pearl-webhooks'


def run_webhook_management_example():
    """Run the webhook management example."""
    if API_KEY == 'YOUR_PEARL_API_KEY':
        print("WARNING: Please replace 'YOUR_PEARL_API_KEY' with your actual API key.")
        print("Exiting example as API Key is a placeholder.")
        return
    
    if YOUR_WEBHOOK_URL == 'https://your-app.com/api/pearl-webhooks':
        print("WARNING: Please replace 'YOUR_WEBHOOK_URL' with your actual webhook endpoint URL.")
        print("Exiting example as Webhook URL is a placeholder.")
        return

    client = PearlClient(api_key=API_KEY)

    try:
        print("\n--- Starting Webhook Management Example ---")

        # --- Register Webhook Endpoint ---
        print(f"Attempting to register webhook endpoint: {YOUR_WEBHOOK_URL}")
        register_request = WebhookEndpointRequest(endpoint=YOUR_WEBHOOK_URL)
        client.webhooks.register(register_request)
        print("Successfully registered webhook endpoint.")

        # --- Update Webhook Endpoint (Example) ---
        # If your webhook URL changes, you can use the update method.
        # For this example, we'll simulate an update to a slightly different path.
        updated_webhook_url = YOUR_WEBHOOK_URL.replace('/pearl-webhooks', '/v2/pearl-webhooks')
        print(f"\nAttempting to update webhook endpoint to: {updated_webhook_url}")
        update_request = WebhookEndpointRequest(endpoint=updated_webhook_url)
        client.webhooks.update(update_request)
        print("Successfully updated webhook endpoint.")

    except Exception as error:
        print("\n--- Webhook Management API Error ---")
        print(f"An error occurred: {error}")
        
        # Additional error handling for requests exceptions
        if hasattr(error, 'response'):
            print(f"Status code: {error.response.status_code}")
            print(f"Response: {error.response.text}")

    finally:
        print("\n--- End of Webhook Management Example ---")


if __name__ == "__main__":
    run_webhook_management_example()
